src/sagas/nanoContract.js

```diff
@@ -213,15 +228,18 @@ export async function fetchHistory(ncId, count, after, wallet) {
       isMine,
       balance,
     };
-    history.push(tx);
+    historyReversed[rawHistory.length - idx - 1] = tx;
   }
 
   let next = after;
-  if (history && history.length > 0) {
-    next = history[history.length - 1].txId;
+  if (historyReversed && historyReversed.length > 0) {
+    // We should get the first item of the list because we are looking
+    // from present -> past, and we want to paginate from preset toward future.
```

Comment:
I'm confused, we're actually paginating towards the past, right?

You scroll to see earlier transactions, i.e. the latest transaction is on top

Reply:
This is confusing, indeed. I believe I can remove this comment. We just need to assign the newest tx hash to `next`, that's it.

Reply:
I have refactored it, can you take a new look?